maven-model-builder/src/main/java/org/apache/maven/model/interpolation/BuildTimestampValueSource.java

```diff
@@ -23,17 +23,19 @@
 import org.codehaus.plexus.interpolation.AbstractValueSource;
 
 class BuildTimestampValueSource extends AbstractValueSource {
-    private final MavenBuildTimestamp mavenBuildTimestamp;
+    private final Date startTime;
+    private final Map<String, String> properties;
 
     BuildTimestampValueSource(Date startTime, Map<String, String> properties) {
```

Comment:
Now that we've switched to Java 8, would it make sense to refactor `Date` to a JSR-310 type?

Reply:
The problem is that the only usage is when using the `ModelBuildingRequest.getBuildStartTime()`, which comes from `ProjectBuildingRequest.getBuildStartTime()` and `MavenExecutionRequest.getStartTime()`.  Without modifying those, which are public interfaces, there's no real value in changing this internal class.
Maybe when the core switches to the v4 api, as the start time is available as an `Instant` on the `Session`, then it will make sense to switch.  For now, it will only add `Date` -> `Instant` conversions for no real value imho.

Reply:
Another possibility is to get rid of the `MavenBuildTimestamp` and simply inline the call inside `BuildTimestampValueSource.getValue()`

Reply:
AFAICS, the `MavenBuildTimestamp` class takes care of properly applying the `maven.build.timestamp.format` value, too. I think it's a bit too much to inline all of that here. Let's leave it as is.